agent/app/agent_capability.go

```diff
@@ -498,6 +503,24 @@ func (agent *ecsAgent) appendServiceConnectCapabilities(capabilities []*ecs.Attr
 	return capabilities
 }
 
+func (agent *ecsAgent) appendGuardDutyCapabilities(capabilities []*ecs.Attribute) []*ecs.Attribute {
+	isEnabled, err := pathExists(guardDutyServiceFilePath, false)
```

Comment:
this check could go in`isGuardDutySupported()` in the `agent_capability_unix.go`. That way we return true if pathExists and append capability - much simpler to read imo.

Reply:
Updated this